beacon/blockchain/finalize_block.go

```diff
@@ -150,6 +150,10 @@ func (s *Service) PostFinalizeBlockOps(ctx sdk.Context, blk *ctypes.BeaconBlock)
 		return fmt.Errorf("sendPostBlockFCU failed: %w", err)
 	}
 
+	// reset latest verified payload in block builder to signal
+	// that no payload is available to reuse for blk.Slot
+	s.localBuilder.CacheLatestVerifiedPayload(blk.Slot, nil)
```

Comment:
This is not strictly necessary, since we do track the slot in block builder, so there is no risk of reusing an already finalized payload (CometBFT won't request a node to rebuild a finalized block).
Still I like the idea of having a clean lifetime for the cached payload, and this clearly signal the one case where the cached payload should be cleared out.